File: 0037-sudoku-solver/0037-sudoku-solver.py
class Solution:
    def check(self, board, row, col, ch):
        for i in range(9):
            if board[row][i] == ch or board[i][col] == ch or board[(row//3)*3 + i//3][(col//3)*3 + i%3] == ch:
                return False
        return True

    def backTracking(self, board):
        for i in range(9):
            for j in range(9):
                if board[i][j] == '.':
                    for ch in '123456789':
                        if self.check(board, i, j, ch):
                            board[i][j] = ch
                            if self.backTracking(board):
                                return True
                            board[i][j] = '.'
                    return False
        return True

    def solveSudoku(self, board: List[List[str]]) -> None:
        """
        Do not return anything, modify board in-place instead.
        """
        self.backTracking(board)

File: LeetCode/0037-sudoku-solver/0037-sudoku-solver.java
class Solution {
    public boolean check(char[][] board, int row, int col, char ch) {
        for(int i=0; i<9 ; i++) {
            if(board[row][i] == ch) {
                return false;
            }
            if(board[i][col] == ch) {
                return false;
            }
            if(board[(row/3)*3+i/3][(col/3)*3+i%3] == ch) {
                return false;
            }
        }
        return true;
    }

    public boolean backTracking(char[][] board) {
        for(int i=0; i<9; i++) {
            for(int j=0; j<9; j++) {
                if(board[i][j] == '.') {
                    for(char ch='1'; ch<='9'; ch++) {
                        if(check(board,i,j,ch)) {
                            board[i][j] = ch;
                            if(backTracking(board) == true) {
                                return true;
                            }
                            board[i][j] = '.';
                        }
                    }
                    return false;
                }
            }
        }
        return true;
    }

    public void solveSudoku(char[][] board) {
        backTracking(board);
    }
}